### PR TITLE
[SPARK-35647][PYTHON][DOCS] Restructure User Guide in PySpark documentation

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to restructure User Guide in PySpark documentation for pandas APIs on Spark.

**Before**

Screen Shot 2021-06-08 at 8 47 41 PM

**After**

Screen Shot 2021-06-08 at 8 46 58 PM

Note that I mostly just moved the contents around except minor changes:
- Removing some questions in FAQ that don't make sense in Apache Spark
- Rename a subtitle "Working with pandas and PySpark" to "From/to pandas and PySpark DataFrames"

For renaming Koalas to either pandas-on-Spark or pandas APIs on Spark, it will be done at SPARK-35591

### Why are the changes needed?

For better readability.

### Does this PR introduce _any_ user-facing change?

Yes, it restructures the documentation as shown above.

### How was this patch tested?

I manually built the docs and tested.

### CODE REVIEW

File: python/docs/source/user_guide/pandas_on_spark/pandas_pyspark.rst [hunk 1]
```diff
@@ -1,6 +1,6 @@
-===============================
-Working with pandas and PySpark
-===============================
+=====================================
+From/to pandas and PySpark DataFrames
```
Comment (xinrong-databricks) [inline python/docs/source/user_guide/pandas_on_spark/pandas_pyspark.rst:2]: `From/to pandas and PySpark`? `From/to pandas/PySpark`? `Conversion from/to pandas and PySpark`? `DataFrames conversion from/to pandas and PySpark`?
I just wanted to provide more options, the existing one looks good as well. :)

**Review by xinrong-meng — APPROVED** [review]

**Comment by xinrong-databricks** [comment]
LGTM! Thanks.

File: python/docs/source/user_guide/pandas_on_spark/pandas_pyspark.rst (hunk 1 above)
Comment (HyukjinKwon) [reply python/docs/source/user_guide/pandas_on_spark/pandas_pyspark.rst:2]: lol. I think it's fine either way. I intentionally wrote PySpark and pandas **DataFrame**s becuase we're already in the same PySpark.

**Comment by HyukjinKwon** [comment]
Thanks @xinrong-databricks.

Merged to master.

**Pull request closed without merging**